src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/RunSelectedTestAction.java

```diff
@@ -0,0 +1,253 @@
+   * @param containingFile {@link VirtualFile} the file that containing the impacted tests.
+   * @param buckFile {@link VirtualFile} the file representing the buck File.
+   * @param runInDebug a {@link boolean} used to signify run or debug.
+   */
+  void createTestConfigurationFromContext(
+      String name,
+      String testSelectors,
+      final Project project,
+      VirtualFile containingFile,
+      VirtualFile buckFile,
+      boolean runInDebug) {
+    BuckBuildManager buildManager = BuckBuildManager.getInstance(project);
+    BuckQueryCommandHandler handler =
+      new BuckQueryCommandHandler(
+          project,
+          buckFile.getParent(),
+          BuckCommand.QUERY, new Function<List<String>, Void>() {
+        @Nullable
+        @Override
+        public Void apply(@Nullable List<String> strings) {
+          String testTarget = strings.get(0);
+          if (!testTarget.isEmpty()) {
+            RunManagerImpl runManager = (RunManagerImpl) RunManager.getInstance(project);
+            TestConfigurationType type = new TestConfigurationType();
+            RunnerAndConfigurationSettingsImpl runnerAndConfigurationSettings = (RunnerAndConfigurationSettingsImpl)runManager.createRunConfiguration(name, type.getConfigurationFactories()[0]);
+            TestConfiguration testConfiguration = (TestConfiguration) runnerAndConfigurationSettings.getConfiguration();
+            testConfiguration.data.target = testTarget;
+            testConfiguration.data.testSelectors = testSelectors;
+            runManager.addConfiguration(runnerAndConfigurationSettings, false);
+            runManager.setSelectedConfiguration(runnerAndConfigurationSettings);
+            if (runInDebug) {
+              ProgramRunnerUtil.executeConfiguration(
+                  project,
+                  runnerAndConfigurationSettings,
+                  DefaultDebugExecutor.getDebugExecutorInstance());
+            } else {
+              Executor executor = ExecutorRegistry.getInstance().getRegisteredExecutors()[0];
+              ProgramRunnerUtil.executeConfiguration(
+                  project,
+                  runnerAndConfigurationSettings,
+                  executor);
+            }
+          }
+          return null;
+        }});
+    handler.command().addParameter("owner("+ containingFile.getPath()+")");
+    buildManager.runBuckCommand(handler, "Getting Selected Test info.");
+  }
+
+  /**
+   * Check that a method is annotated @Test.
+   *
+   * @param method a {@link PsiMethod} to check.
+   * @return {@code true} if the method has the junit Test annotation. {@code false} otherwise.
+   */
+  private boolean isTestMethod(PsiMethod method) {
+    boolean isTestClass = false;
+    method.getModifierList().getAnnotations();
+    PsiAnnotation[] annotations = method.getModifierList().getAnnotations();
+    for (PsiAnnotation annotation : annotations) {
+      if (ORG_JUNIT_TEST_ANNOTATION.equals(annotation.getQualifiedName())) {
```

Comment:
What about TestNg ? If it is not supported yet (buck supports test ng), please make a note in the documentation of this class